src/AdminApp/ManageSalons.js

```diff
@@ -0,0 +1,249 @@
+            },
+          }
+        );
+        let json = [];
+
+        if (response.ok) {
+          json = await response.json();
+          setSalons(json);
+        } else {
+          alert("Nie znaleziono rekordow  w bazie danych dla 1 " + category);
+        }
+        setSalons(json);
+        console.log(json);
+        console.log("Kategory i bounds");
+      }
+    }
+
+    load();
+  }, [category]);
+
+  useEffect(() => {
+    async function load() {
+      console.log(bbox);
+      if (category !== null && bbox !== null) {
+        const response = await fetch(
+          `http://localhost:8080/services/${category}`,
+          {
+            method: "POST",
+            body: JSON.stringify({
+              serviceName: category,
+              LatBtm: bbox.btmRightPoint?.lat,
+              LonBtm: bbox.btmRightPoint?.lon,
+              LatTop: bbox.topLeftPoint?.lat,
+              LonTop: bbox.topLeftPoint?.lon,
+              distance: 0,
+              treatment: null,
+              priceFrom: 0,
+              priceTo: 0,
+              opinion: 0.0,
+              localizationLat: latitude,
+              localizationLon: longitude,
+            }),
+            headers: {
+              "Content-Type": "application/json",
+            },
+          }
+        );
+        let json = [];
+
+        if (response.ok) {
+          json = await response.json();
+          setSalons(json);
+        } else {
+          alert("Nie znaleziono rekordow  w bazie danych dla " + category);
+        }
+        setSalons(json);
+        console.log("Kategory i bbox");
+      }
+    }
+    load();
+  }, [category, bbox]);
+
+  //===============================
+  useEffect(() => {
+    async function load() {
+      console.log(bbox);
+      if (category !== null && bbox == null && salons.length === 0) {
+        const response = await fetch(
+          `http://localhost:8080/bounds-services/${category}`,
+          {
+            method: "POST",
+            body: JSON.stringify({
+              serviceName: category,
+              LatBtm: bounds._ne.lat,
+              LonBtm: bounds._sw.lng,
+              LatTop: bounds._sw.lat,
+              LonTop: bounds._ne.lng,
+              distance: 0,
+              treatment: null,
+              priceFrom: 0,
+              priceTo: 0,
+              opinion: 0.0,
+              localizationLat: latitude,
+              localizationLon: longitude,
+            }),
+            headers: {
+              "Content-Type": "application/json",
+            },
+          }
+        );
+        let json = [];
+
+        if (response.ok) {
+          json = await response.json();
+          setSalons(json);
+        } else {
+          alert("Nie znaleziono rekordow  w bazie danych dla " + category);
+        }
+
+        console.log(json);
+        console.log("Salons i category i bounds");
+      }
+    }
+    load();
+  }, [salons]);
+
+  useEffect(() => {
```

Comment:
Two `useEffects` are fired by the change of the `salons` and each of them are invoking `setSalons` at the end.  I see that they probalby have different data in bodies, but one of them is overriding the second's `setSalons` (as both have the same condition). And the dependency array of the `useEffect` contains only `salons` that is changed in this `useEffect`. If `salons` won't be changed the useEffect won't be fired anymore. Maybe `category` or `bbox` is better if it changes? ;)